www/js/lib/zimfile.js

```diff
@@ -20,7 +20,7 @@
  * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
  */
 'use strict';
-define(['xzdec_wrapper', 'util', 'utf8', 'q'], function(xz, util, utf8, Q) {
+define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util, utf8, Q, zimDirEntry) {
```

Comment:
I don't master this code portion (I did not write it myself).
I agree with you that it lacks some comments. And it might be a place where some optimization could take place.
I'll leave it as it is for now.